public/app/features/api-keys/ApiKeysTable.tsx

```diff
@@ -44,7 +46,12 @@ export const ApiKeysTable: FC<Props> = ({ apiKeys, timeZone, onDelete }) => {
                   )}
                 </td>
                 <td>
-                  <DeleteButton aria-label="Delete API key" size="sm" onConfirm={() => onDelete(key)} />
+                  <DeleteButton
+                    aria-label="Delete API key"
+                    size="sm"
+                    onConfirm={() => onDelete(key)}
+                    disabled={!canDelete}
```

Comment:
It could be the case that the user is only allowed to delete some of the keys. In which case if we want to do this properly, we should attach access control metadata to the returned API keys and check whether the user is allowed to delete the specific key (we've done this for teams, data sources and some other resources).

However, I think it's fine to leave this as it is, as this is a minor issue and as we're deprecating API keys relatively soon.

Reply:
Really good point, I am 50/50 about if I should this now or postpone, given the timeline it would probably make sense to include this with the next release and not push hard for 8.5. Here is an [issue](https://github.com/grafana/grafana-enterprise/issues/3177) to track the work.